packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-spread/expression/expected.js

```diff
@@ -1,9 +1,23 @@
-function _extendsDefine() { _extendsDefine = function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; Object.getOwnPropertyNames(source).forEach(function (propKey) { var desc = Object.getOwnPropertyDescriptor(source, propKey); Object.defineProperty(target, propKey, desc); }); if (Object.getOwnPropertySymbols) { Object.getOwnPropertySymbols(source).forEach(function (symbKey) { var desc = Object.getOwnPropertyDescriptor(source, symbKey); Object.defineProperty(target, symbKey, desc); }); } } return target; }; return _extendsDefine.apply(this, arguments); }
+function _spreadByDefine(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { var desc = Object.getOwnPropertyDescriptor(source, key); Object.defineProperty(target, key, desc); } } } return target; }
 
-_extendsDefine({
+_spreadByDefine({
   x
 }, y, {
   a
 }, b, {
   c
 });
+
+_spreadByDefine({}, Object.prototype);
+
+_spreadByDefine({}, {
```

Comment:
This test (and the one below) are failing... As you can see is not being transformed correctly: the first parameter is omitted causing the following result:

```js
_spreadByDefine({
  foo: 'bar'
});
```

I'd love if someone can point me at the right direction because right now I don't know how to fix this one. I guess the problem is in the plugin itself? `babel-plugin-proposal-object-rest-spread/src/index.js`

Reply:
Instead of adding the object expression only if the first argument isn't an object expression, you should add it before the loop if the first prop is a spread property.

https://github.com/nuragic/babel/blob/71d8313fef584ad7ff8bb7a7d5e79e6599d60df3/packages/babel-plugin-proposal-object-rest-spread/src/index.js#L370-L372